backend/src/main/controller/PromotionController.ts

```diff
@@ -0,0 +1,121 @@
+import { NextFunction, Request, Response } from 'express';
+import { PromotionRepository } from '../repository/PromotionRepository';
+import { getCustomRepository } from 'typeorm';
+import { Promotion } from '../entity/Promotion';
+import { UserRepository } from '../repository/UserRepository';
+import { Discount } from '../entity/Discount';
+import {
+  PromotionDTO,
+  PromotionValidation,
+} from '../validation/PromotionValidation';
+import { IdValidation } from '../validation/IdValidation';
+
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+export class PromotionController {
+  /**
+   * Retrieves all promotions and their discounts
+   * * First we need to validate the query params and cast that into a PromotionQueryDTO
+   * * Then we apply the query options into the query builder depending on which properties are present
+   * * Now we execute the query builder and return its results
+   */
+  getAllPromotions = async (
+    request: Request,
+    response: Response,
+    next: NextFunction
+  ): Promise<any> => {
+    try {
+      const promotions = await getCustomRepository(
+        PromotionRepository
+      ).getAllPromotions(request.query);
```

Comment:
@andrewzulaybar 
This is kind of what the search query is like atm. You can add a number of request query params and they're all an AND clause. I plan on working a bit more on searching by `name`, but this is what I have so far.

![image](https://user-images.githubusercontent.com/49849754/99158624-a2e19580-2689-11eb-824c-0f175852cf33.png)